### PR TITLE
build: update to Typescript 3.6

### PR DESCRIPTION
**Pull request by jelbourn**

### CODE REVIEW

File: src/material/menu/menu.spec.ts [hunk 1]
```diff
@@ -1071,8 +1071,8 @@ describe('MatMenu', () => {
       // In "before" position, the right sides of the overlay and the origin are aligned.
       // To find the overlay left, subtract the menu width from the origin's right side.
       const expectedLeft = triggerRect.right - overlayRect.width;
-      expect(Math.floor(overlayRect.left))
-          .toBe(Math.floor(expectedLeft),
+      expect(Math.abs(Math.floor(overlayRect.left) - Math.floor(expectedLeft)))
```
Comment (mmalerba) [inline src/material/menu/menu.spec.ts]: Is this related to TS update or just fixing flaky tests while you're here?

File: src/cdk/clipboard/pending-copy.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ * forces a relayout. This relayout can take too long if the string is large,
+ * causing the execCommand('copy') to happen too long after the user clicked.
+ * This results in the browser refusing to copy. This object lets the
+ * relayout happen in a separate tick from copying by providing a copy function
+ * that can be called later.
+ *
+ * Destroy must be called when no longer in use, regardless of whether `copy` is
+ * called.
+ */
+export class PendingCopy {
```
Comment (mmalerba) [inline src/cdk/clipboard/pending-copy.ts]: Is moving this to a separate file related to updating TS? (ok leaving it here, just curious)

File: src/cdk/overlay/overlay-config.ts [hunk 3]
```diff
@@ -70,7 +74,7 @@ export class OverlayConfig {
           // is not smart enough to see that the right-hand-side is actually an access of the same
           // exact type with the same exact key, meaning that the value type must be identical.
           // So we use `any` to work around this.
-          this[key] = config[key] as any;
+          this[key] = (config)[key] as any;
```
Comment (mmalerba) [inline src/cdk/overlay/overlay-config.ts]: Seems unnecessary?

File: package.json [hunk 4]
```diff
@@ -35,39 +35,40 @@
     "format:bazel": "yarn -s bazel:buildifier --lint=fix --mode=fix",
     "format": "yarn -s format:ts && yarn -s format:bazel",
     "cherry-pick-patch": "ts-node --project tools/cherry-pick-patch/ tools/cherry-pick-patch/cherry-pick-patch.ts",
-    "check-rollup-globals": "ts-node --project scripts/ scripts/check-rollup-globals.ts"
+    "check-rollup-globals": "ts-node --project scripts/ scripts/check-rollup-globals.ts",
+    "debug": "ngc -p ./src/cdk/coercion/tsconfig-build.json"
```
Comment (devversion) [inline package.json]: Should we remove this?

Comment (jelbourn) [reply package.json]: Done, leftover from debugging

File: tools/BUILD.bazel [hunk 5]
```diff
@@ -1,28 +1,5 @@
 package(default_visibility = ["//visibility:public"])
 
-load("@build_bazel_rules_nodejs//:defs.bzl", "nodejs_binary")
-load("//tools:defaults.bzl", "ts_library")
-
-ts_library(
-    name = "sass_bundle_lib",
-    srcs = ["sass-bundle.ts"],
```
Comment (Splaktar) [inline tools/BUILD.bazel]: How are we handling the Sass bundling and copying now that this is removed?

Are we still using the `scss-bundle` dependency?

Comment (jelbourn) [reply tools/BUILD.bazel]: rules_nodejs has a new feature that lets you use a npm package from node_modules without setting up extra rules

https://github.com/angular/components/blob/29be7686efa55d9027db08c50717789b23021a78/src/material/BUILD.bazel#L24-L34

Comment (Splaktar) [reply tools/BUILD.bazel]: Ah, that's nice!

**Pull request merged**